### DISCUSSION TITLE
Truncated test name on the XML report

### DISCUSSION THREAD

**Discussion by Roberta-FCorrea** [xunit/xunit#2582] [Question] [answered]
The complete test name with arguments when executing dotnet test looks like this:
_[xUnit.net 00:01:43.16]     Successfully commence lease by running a Bookie workflow(BusinessUnit: "Default", businessUnit: "707", vendorPartyNumber: "valid", propertyTaxResponsibility: "DoNotRemit", originationSourceType: "Direct", dealType: "Lease", transactionType: "FMV", lineOfBusiness: "Canada", branch: "666457", invoiceFormat: "English", invoiceLeadDays: "25", invoiceDueDay: "30", contractType: "TL", contractTerm: "12", contractPayment: "96.99", contractCommecementDate: "11/30/2017", legalEntity: "Canada", currency: "CAD", assetBookingCurrency: "CAD", invoiceDate: "11/09/2017", invoiceDueDate: "12/23/2017", equipmentDescription: "Inspiron 5680", assetDeliveryDate: "11/16/2017", equipmentAddress1: "760 HWY BEDFORD", equipmentAddress2: "", equipmentCity: "BEDFORD", equipmentState: "NS", equipmentZip: "B4A 3Z7", countryCode: "CAN", equipmentCost: "300", propertyTaxCost: "0.0", lateFeeTemplate: "CAD Daily Late Fee 1.5%", residualAmount: "0", rPAddressLine1: "", rPCity: "", rPZip: "", rPName: "", tapeAccountNumber: "", paymentsArrears: "0", firstPaymentAmount: "21", hasBlendedItems: "false", numberOfPayments: "12", customerCostCenter: "ccc-fit-no-bi", isBargainPurchaseOption: "false", isTaxLease: "true", exampleTags: []) [FAIL]_

In the Xunit report the test name attribute is truncated like this:
_<test name="Successfully commence lease by running a Bookie workflow(BusinessUnit: &quot;Default&quot;, businessUnit: &quot;707&quot;, vendorPartyNumber: &quot;valid&quot;, propertyTaxResponsibility: &quot;DoNotRemit&quot;, originationSourceType: &quot;Direct&quot;, dealType: &quot;Lease&quot;, transactionType: &quot;Dollar1Out&quot;, lineOfBusiness: &quot;Canada&quot;, branch: &quot;666457&quot;, invoiceFormat: &quot;English&quot;, invoiceLeadDays: &quot;25&quot;, invoiceDueDay: &quot;30&quot;, contractType: &quot;LP&quot;, contractTerm: &quot;12&quot;, contractPayment: &quot;96.99&quot;,..."_

Is there a way to have the entire test name with its arguments in the test name XML report?

**Comment by Roberta-FCorrea** [comment]
Found the code where xUnit runner truncates the test name:
`const string Ellipsis = "...";`
`const int MaximumDisplayNameLength = 447;`

applied here:
```static string Truncate(string value)
        {
            if (value.Length <= MaximumDisplayNameLength)
                return value;

            return value.Substring(0, MaximumDisplayNameLength - Ellipsis.Length) + Ellipsis;
        }

**Comment by Roberta-FCorrea** [comment]
Couldn't it become configurable?

**Comment by bradwilson** [comment, marked as answer]
There is an open issue for this: #1914
